src/Pages/Repositories/PopularRepositoriesTable/components/AddRepo.tsx

```diff
@@ -42,20 +44,64 @@ export const AddRepo = ({ isDisabled, addRepo }: Props) => {
     onActionFocus();
   };
 
-  const dropdownItems = [
-    <DropdownItem
-      data-ouia-component-id='add-popular_repo_without-snapshotting'
-      key='action'
-      component='button'
-      onClick={() => addRepo(false)}
-    >
-      Add without snapshotting
-    </DropdownItem>,
-  ];
+  const dropdownItems = (isProd: boolean) => {
+    if (isProd) {
```

Comment:
This all looks pretty good, what I'd suggest is checking isProd once with a useMemo above, as that value shouldn't change. 

Then using a constant for that value within the render, or a useMemo on this function with the isProd constant as a dependency.

The reason you'd make this change is to prevent unneeded re-renders of this component. 
Granted in this case, the total performance loss is likely very very small; the concept is more what I'm after for you to practice here, as performance difference on larger pages, or repeated components can be significant.

You can test the result of this described change by adding a console.log to your dropdownItems function here, or adding a count variable to the context, and allowing each of the dropdowns to trigger a single update of that count. This can make the results of the change much easier to see.

With the described method you should see far less re-renders.

Reply:
thanks! using a useMemo, it looks like the amount of renders were reduced by about 5 times. let me know if my changes are what you meant